compiler/rustc_lint/src/context/diagnostics.rs

```diff
@@ -261,17 +278,30 @@ pub(super) fn builtin(
                 }
 
                 is_feature_cfg |= best_match == sym::feature;
-            } else if !possibilities.is_empty() {
-                let mut possibilities =
-                    possibilities.iter().map(Symbol::as_str).collect::<Vec<_>>();
-                possibilities.sort();
-                let possibilities = possibilities.join("`, `");
+            } else {
+                if !names_possibilities.is_empty() {
+                    names_possibilities.sort();
+                    for cfg_name in names_possibilities.iter() {
+                        db.span_suggestion(
```

Comment:
We should limit the amount of spanned suggestion since I can think of real cases where the number could be huge (an example could be in the Rust-for-Linux project where someone writes `y` as cfg name, there could be hundreds, maybe thousand of valid suggestion: `CONFIG_NVME = "y"`, `CONFIG_SAS = "y"`, ...).

I would suggest making the limit at 3.
```suggestion
                if !names_possibilities.is_empty() && names_possibilities.len() <= 3 {
                    names_possibilities.sort();
                    for cfg_name in names_possibilities.iter() {
                        db.span_suggestion(
```

Reply:
good idea.